restructure CLAUDE.md (#382)
posit-dev/ggsql: tree-sitter-ggsql/CLAUDE.md

│   └── c/                  C binding glue
├── queries/
│   └── highlights.scm      Highlight queries used by editors
└── test/
    └── corpus/             Tree-sitter test corpus (parser test cases)
```

The files under `src/` are generated by `tree-sitter generate` from `grammar.js` — they are committed so downstream consumers don't need the tree-sitter CLI to build.

## Grammar at a glance

`grammar.js` defines a `query` rule that is `optional(sql_portion) + repeat(visualise_statement)`. The SQL portion is recognised structurally enough to know where it ends (statement boundaries, recursive subqueries, WITH compound statements) without re-implementing a SQL parser — the actual SQL is handed off to the configured `Reader`. The VISUALISE side is parsed in detail (clauses, layer types, mappings, settings, scales, facets, projections, labels) so [`/src/parser/builder.rs`](../src/parser/builder.rs) can build a typed AST from it.

For ggsql language semantics, see [`/doc/syntax/`](../doc/syntax/) — this package only defines *how text is parsed*, not what the resulting tree means.

## Workflow

```sh
# Regenerate src/grammar.json + parser.c after editing grammar.js

> npx tree-sitter generate
> 
> # Run the corpus tests in test/
> npx tree-sitter test

Reviewer: ```suggestion
tree-sitter generate

# Run the corpus tests in test/
tree-sitter test
```
Windows thingy again

# All tests via npm (corpus + the Node binding smoke test)
npm test
```

> In editors with tree-sitter support: `npx tree-sitter playground` to interactively probe the parser.

Reviewer: ```suggestion
In editors with tree-sitter support: `tree-sitter playground` to interactively probe the parser. If needed, the tree-sitter WASM may be rebuilt by using `tree-sitter build --wasm`.
```

## Adding new ggsql syntax

1. Edit `grammar.js`. Pick node names that downstream code can match against.
2. Add corpus tests under `test/corpus/` covering the new construct (and any conflicts).

> 3. `npx tree-sitter generate` to regenerate `src/grammar.json` and `parser.c`. Commit those.

Reviewer: ```suggestion
3. `tree-sitter generate` to regenerate `src/grammar.json` and `parser.c`. Commit those.
```

4. Update [`/src/parser/builder.rs`](../src/parser/builder.rs) to produce the matching AST nodes.

> 5. If the new syntax should be highlighted, update `queries/highlights.scm`.

Reviewer: I thought we had more highlight files/autocomplete files that need to recognise some syntax. Is this the only one we have left now that other modules have moved out?
